internal/common/db.go

```diff
@@ -145,6 +145,11 @@ func GetClustersFromDBWhere(db *gorm.DB, eagerLoading EagerLoadingState, include
 	if err != nil {
 		return nil, err
 	}
+	// we need to fetch Hosts association to allow AfterFind hook to run
+	for _, cluster := range clusters {
```

Comment:
I think that `GetClustersFromDBWhere` as a utility shouldn't define this logic.
What about calling it on `inventory.go` with `SkipEagerLoading`?

Reply:
This would prevent loading `MonitoredOperators` which are prefetched as well. I am not sure whether UI is using it or not so I did not want to do it. With `AfterFind` we still need to query for the hosts :/

Reply:
I think that monitored operators aren't used, but we need the hosts anyhow.
Could you move the logic of emptying the hosts' field to the calling function on `inventory.go`? So the utility will still be useable

Reply:
Sure, will move

Reply:
@pkliczewski  You forgot to delete it from here :)

Reply:
oh :/